Upgrade to Micronaut 3

cemo opened the pull request:
This PR is sent separated to address #577.

cemo commented:
The PR is failing on my local environment. I am trying to understand what is wrong but the response 

` $ref: #/components/schemas/MutableHttpResponse_Object_` 

It seems it is not working properly on 3.x. I am looking for it now.

ilopmar commented:
@cemo You need to convert the project to be Micronaut 3 compatible, check this comment: https://github.com/micronaut-projects/micronaut-core/issues/5607#issue-922763986

cemo commented:
It was actually a just a simple PR to address issue we hit. Now I am in the process of migrating to 3.x. 😂

cemo commented:
@ilopmar I checked that PR. This project somehow has already done these changes but it is only missing these props I changed. Changing these props is causing a test case to fail. Would you please provide necessary guidance about this project where should I take a look?

ilopmar commented:
The tests are failing because of missing RxJava2 classes, you need to add the dependency:

```groovy
testImplementation "io.reactivex.rxjava2:rxjava"
```

Also it is probably a good idea to add some test that uses `Mono` instead of `Single`/`Maybe`. For that you also need to add Project Reactor core dependency.

cemo commented:
@ilopmar 

When I debug, I see that the implementation has changed slightly.

https://github.com/micronaut-projects/micronaut-openapi/blob/aab3652dd723d48b569ad9a8e3f40e673099b489/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java#L1360-L1364

in 2.x returns "Object"
in 3.x returns "MutableHttpResponse_Object_"

cemo commented:
https://github.com/micronaut-projects/micronaut-openapi/blob/aab3652dd723d48b569ad9a8e3f40e673099b489/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java#L1346-L1358

How should be the correct implementation of this with 3.x? 
For `io.micronaut` prefixed packages returning `Object` sounds ridiculous? 

cc @graemerocher

cemo commented on openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiEndpointVisitor.java:
```diff
@@ -888,9 +891,15 @@ private boolean isResponseType(ClassElement returnType) {
-                && returnType.getFirstTypeArgument().isPresent()
-                && isResponseType(returnType.getFirstTypeArgument().get());
+        boolean assignable = returnType.isAssignable("io.reactivex.Single") ||
+                             returnType.isAssignable("org.reactivestreams.Publisher");
```
This part was the actual problem

cemo commented:
@ilopmar I addressed remaining issues. I think we are ready to go. Would you please review it? I will send another PR to address another issue as well after it.

ilopmar commented:
Thanks for your contribution!
